editor-components/feature/feature9/feature9.tsx

```diff
@@ -1,192 +1,280 @@
+              value:
+                "By leveraging the latest technologies, we share your brand and products with the world while focusing on a perfect visual output through stable and high-performing code. Thats all .",
+              displayer: "Description",
             },
             {
               type: "image",
               key: "image",
               value:
-                "https://storage.googleapis.com/download/storage/v1/b/hq-composer-0b0f0/o/644a673cf72de2002caaa35f?alt=media&timestamp=1682597705111",
+                "https://storage.googleapis.com/download/storage/v1/b/hq-composer-0b0f0/o/66a75c352f8a5b002ce6d211?alt=media",
               displayer: "Image",
             },
           ],
         },
       ],
     });
-    this.addProp({
-      type: "number",
-      key: "itemCount",
-      displayer: "Item count in a row",
-      value: 3,
-    });
-  }
 
+    
+  }
   getName(): string {
     return "Feature 9";
   }
 
+
+
+  setupObservers() {
+    const featureItems = this.castToObject<FeatureItem[]>("feature-items");
+    featureItems.forEach((item, index) => {
+      const element = document.getElementById(`card-${index}`);
+      if (element) {
+        this.observers[index] = new IntersectionObserver(
+          ([entry]) => this.handleIntersection(entry, index),
+          { threshold: 0.82 }
+        );
+        this.observers[index].observe(element);
+      } else {
+        console.log(`Element not found for card-${index}`);
+      }
+    });
+  }
+  handleIntersection(entry: IntersectionObserverEntry, index: number) {
+    const counterElement = document.getElementById("counter");
+    const currentElement = entry.target as HTMLElement;
+    const nextElement = document.getElementById(`card-${index + 1}`);
+
+    if (entry.isIntersecting) {
+      currentElement.style.opacity = '1';
+      this.updateCounter(index + 1);
+      
+      if (nextElement) {
+        nextElement.style.opacity = '0';
+      }
+    } else {
+      counterElement.classList.add(this.decorateCSS("invisible"));
+      currentElement.style.opacity = '0';
+      
+      if (nextElement) {
+        nextElement.style.opacity = '0';
+      }
+    }
+
+    if(currentElement.id=="card-1") {
+      counterElement.classList.remove(this.decorateCSS("invisible"));
+    }
+  }
+
+  updateCounter(id: number) {
+    const counterElement = document.getElementById("counter");
+    if (counterElement) {
+      counterElement.innerText = id.toString();
+      counterElement.classList.remove(this.decorateCSS("invisible"));
+    }
+  }
+
+  renderCard(featureItem: FeatureItem, index: number) {
+    return (
+      <div 
+        key={index} 
+        id={`card-${index}`} 
+        className={this.decorateCSS("card-item")}>
+        <img
+          alt="description image" 
+          className={this.decorateCSS("image")}
+          src={featureItem.image}
+        />
+        <h4 className={this.decorateCSS("sub-title")}>{featureItem.title}</h4>
```

Comment:
don't render h4 if title is empty
[check here](https://github.com/Teknodev/composer-tools/blob/main/editor-components/README.md#how-to-check-if-a-string-prop-is-empty)